components/patient-data/ui/src/main/resources/PhenoTips/CheckID.xml

```diff
@@ -44,11 +44,16 @@
 #if ($request.getParameter('eid') &amp;&amp; $request.getParameter('id'))
   #set ($externalIdentifier = "$!{request.eid}")
   #set ($identifier = "$!{request.id}")
-  #if ($externalIdentifier == $xwiki.getDocument($services.model.resolveDocument($identifier, $services.model.resolveSpace('data'))).getObject('PhenoTips.PatientClass').getProperty('external_id').value)
+  #set ($principal = "$!{request.principal}")
```

Comment:
`principal` is not the right concept here, that's for users and groups. Should be `entity`.